functions/defaulted-parameters.md

```diff
@@ -93,7 +93,7 @@ Subsequent redeclarations of the same function may add default argument
 values, which are then usable by callers.
 Though a single argument cannot be given a default twice in the same 
 translation unit, it is legal, though ill-advised, to give the same
-function different default parameters in different translation units.
+function different default arguments in different translation units.
```

Comment:
This has never been true for inline functions, and is no longer true for any functions: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1766r1.html#default-argument-inconsistency

Reply:
It is permissible to give functions different default arguments in different *scopes* (eg, a block-scope function declaration can have default arguments and those need not match those in any other scope), but that's an expert-level topic.

Reply:
Off-topic for SG20, but: In what sense could the same lexical "scope" span two completely different "translation units"? I don't see any teachable difference between

```
// main.cpp
int foo() { extern int d(int x=1); d(); }
int bar() { extern int d(int x=2); d(); }
```
and
```
// foo.cpp
int d(int x=1); int foo() { d(); }
// bar.cpp
int d(int x=2); int foo() { d(); }
```
I agree that this is an "expert-level topic" and in fact teachers should be very clear with their students that you should _never ever_ give the same function different default arguments. (I simply teach that [default function arguments are the devil](https://quuxplusone.github.io/blog/2020/04/18/default-function-arguments-are-the-devil/), and leave it at that.)

Reply:
I believe this is one of those topics that fall under my category of "Never write this, but you may need to be aware of it when dealing with other developers or long-lived codebases." IIUC, this is in the most "advanced" section we have in our skeleton.  Though now that it is forbidden across translation units, I'd be fine to strike the paragraph, altogether, unless there's something interesting there to keep.

Reply:
@Quuxplusone (Continuing off-topic discussion. Perhaps we should move this to email?)

> In what sense could the same lexical "scope" span two completely different "translation units"?

This can happen via `extern "C"`:
```
namespace A { extern "C" void f(int = 1); }
namespace B { extern "C" void f(int = 2); }
```
... declares the same function with different default arguments in two different scopes, both of which can span multiple translation units.

> I don't see any teachable difference between
> ```
> // main.cpp
> int foo() { extern int d(int x=1); d(); }
> int bar() { extern int d(int x=2); d(); }
> ```
> and
>```
> // foo.cpp
> int d(int x=1); int foo() { d(); }
> ```
> ```
> // bar.cpp
> int d(int x=2); int foo() { d(); }
> ```

Well, the former is valid and the latter is ill-formed (NDR). Whether that qualifies as a teachable difference would depend on the level of the audience, though even at expert level I think it's probably not worth mentioning. "Don't do that" seems like substantially better advice to me, both for declaring the same function with different default arguments, and for block-scope function declarations.